### DISCUSSION TITLE
Github action failed without any log

### DISCUSSION THREAD

**Discussion by cindyyan317** [community/community#52057] [Actions]
### 

Bug

###

Our project has a job to build package on MAC and run unittests. it ran well until the last week. The job failed for each build from time to time and there was no log.
The code can be built successfully on our local machine.

https://github.com/XRPLF/clio/actions/runs/4617650798/jobs/8164153539

Wonder if this is a github action's bug?

**Comment by vitalykarasik** [comment]
I guess it's because some tools/libraries were updated. 
I suggest comparing all versions between your 'good' and 'bad' build environments.

**Reply by cindyyan317** [reply]
Hi @vitalykarasik  Thanks for your reply. It happens quite randomly. like for this PR, https://github.com/XRPLF/clio/pull/577/checks. The first commit's build was failed, but the second one was okay, there was not the related change tho.

**Reply by vitalykarasik** [reply]
I suggest to review failed step logs (unfortunately I couldn't see them).

**Reply by cindyyan317** [reply]
Thanks. 
The only log I can see is "The hosted runner: GitHub Actions 3 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error."

My application is a normal C++ lib+binary, nothing heavy. I guess the runner has been too small

**Reply by vitalykarasik** [reply]
Oh! Not sure that even 'debug mode' will help us. Don't think we have access to managed runners monitoring...
The only way to debug it without GitHub support help, IMHO, it's trying to check on your local machine how much RAM and disk build/test workflow uses. Maybe you underestimated its requirements. Mac hosted runners have 14GB RAM and 14GB disk (of course not all 14GB is free - on linux we have about 8GB free, and we can clean more).
